src/test/ui/span/issue-42595.stderr

```diff
@@ -0,0 +1,16 @@
+error[E0368]: binary assignment operation `+=` cannot be applied to type `&mut {integer}`
+  --> $DIR/issue-42595.rs:15:5
+   |
+15 |     map.entry("e").or_insert(0) += 1;
+   |     ---------------------------^^^^^
+   |     |
+   |     cannot use `+=` on type `&mut {integer}`
```

Comment:
Oh. I would have expected the span to be on `+=`, actually, not the whole expression or some part of it. Obviously that's a separate discussion though -- no need to fix here.